src/lib.rs

```diff
@@ -672,21 +668,21 @@ impl GenerateTasksService for GenerateTasksServiceImpl {
             let bv_info = burn_in_tag_build_variant_info
                 .entry(variant.clone())
                 .or_insert(BurnInTagBuildVariantInfo {
-                    compile_distro: compile_distro.clone(),
                     compile_task_group_name: compile_task_group_name.clone(),
+                    compile_variant: compile_variant.clone(),
                 });
-            if bv_info.compile_distro != compile_distro.clone() {
-                panic!(
-                    "`{}` is trying to set a different compile distro than already exists for `{}`. Check the `{}` expansions in your config.",
-                build_variant.name, variant, BURN_IN_TAG_COMPILE_DISTRO
-            )
-            }
             if bv_info.compile_task_group_name != compile_task_group_name {
                 panic!(
                     "`{}` is trying to set a different compile task group name than already exists for `{}`. Check the `{}` expansions in your config.",
                 build_variant.name, variant, BURN_IN_TAG_COMPILE_TASK_GROUP_NAME
             )
             }
+            if bv_info.compile_variant != compile_variant {
+                panic!(
+                    "`{}` is trying to set a different compile variant than already exists for `{}`. Check the `{}` expansions in your config.",
+                    build_variant.name, variant, COMPILE_VARIANT
```

Comment:
Could there be cases where we would want to have a dedicated `burn_in_tag_compile_variant` expansion instead of using a single `compile_variant` expansion for the current build variant and also burn_in_tag build variants it produces?

I assume it should be easier to understand how to configure burn_in_tags or what is wrong with configuration with dedicated expansions.

Reply:
Good question Misha. For this specific option, there's always been a desire to remove the dedicated compile variant for burn in, which is done as a part of this ticket. 

Longer-term, the plan is to move the really non-deterministic testing to mainline only so burn in doesn't become the scapegoat for all flaky test issues.

Reply:
All right, sounds good to me!